contrib/mesos/pkg/scheduler/service/service.go

```diff
@@ -459,9 +467,12 @@ func (s *SchedulerServer) prepareExecutorInfo(hks hyperkube.Interface) (*mesos.E
 
 	if s.kubeletKubeconfig != "" {
 		//TODO(jdef) should probably support non-local files, e.g. hdfs:///some/config/file
-		uri, basename := s.serveFrameworkArtifact(s.kubeletKubeconfig)
-		ci.Uris = append(ci.Uris, &mesos.CommandInfo_URI{Value: proto.String(uri)})
-		ci.Arguments = append(ci.Arguments, fmt.Sprintf("--kubeconfig=%s", basename))
+		if s.kubeletKubeconfig != s.proxyKubeconfig {
+			// allows kubelet-kubeconfig and proxy-kubeconfig to point to the same file
```

Comment:
Good point! Perhaps instead of using the basename, we can use the absolute path of the file with `s/\///g`, or a hash calculated based on the absolute path of the file. However, it could look ugly when you do a `ps aux | grep km` on the mesos node running `km [executor|proxy]`. It might be an overkill here. I still favor a friendly message that says "different kubeconfig files should have different basenames". If this is OK, I'll make this change and do another push.